spec/util/java_version_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'spec_helper'
+require 'logstash/util/java_version'
+
+describe "LogStash::Util::JavaVersion" do
+  let(:mod) { LogStash::Util::JavaVersion }
+  it "should get the current java version if we're on Java" do
+    if RUBY_ENGINE == "jruby"
+      expect(LogStash::Util::JavaVersion.version).to be_a(String)
+    end
+  end
+
+  it "should mark a bad java version as bad" do
+    expect(mod.bad_java_version?("1.7.0_45-beta")).to be_truthy
```

Comment:
Here, you have at the end two test cases, one with the beta and another without. This should be two it section, so when debugging we get a nice error message.